Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs

```diff
@@ -150,9 +149,9 @@ public void Populate(List<VendingMachineInventoryEntry> inventory, float priceMo
                 }
 
                 // This block exists to allow the VendPrice flag to set a vending machine item price.
-                if (prototype != null && prototype.TryGetComponent<VendPriceComponent>(out var vendPriceComponent) && vendPriceComponent.Price != 0 && cost <= (float) vendPriceComponent.Price)
+                if (prototype != null && prototype.TryGetComponent<StaticPriceComponent>(out var vendPriceComponent) && vendPriceComponent.VendPrice != 0 && cost <= (float) vendPriceComponent.VendPrice)
                 {
-                    var price = (float) vendPriceComponent.Price;
+                    var price = (float) vendPriceComponent.VendPrice;
```

Comment:
Might be a dumb question, but why have the check against static price?  I understand that not having it could make a vendor sell things at a discount compared to its base, but if you want it overridden, should it not be overridden?

Reply:
If you set static price + multi to lower then vend. Vend need to be ignored